torch/onnx/_internal/exporter/_registration.py

```diff
@@ -274,5 +280,19 @@ def is_registered(self, target: TorchOp) -> bool:
         """
         return bool(self.get_decomps(target))
 
+    def _cleanup_registry_based_on_opset_version(self) -> None:
+        """Pick the implementation with the highest opset version valid until the current opset version."""
+        cleaned_functions = {}
+        for target_or_name, decomps in self.functions.items():
+            # Filter decompositions to only include those with opset_introduced <= opset_version
+            decomps = [d for d in decomps if d.opset_introduced <= self.opset_version]
+
+            # Keep only the decomposition with the highest opset_introduced
+            if decomps:
+                max_opset_decomp = max(decomps, key=lambda x: x.opset_introduced)
+                cleaned_functions[target_or_name] = [max_opset_decomp]
```

Comment:
Hmm, this is not right. There may be multiple decomps we want to keep. (Ref: the failing complex test)

Reply:
I think something like the following will make sense:

if there are decomps for multiple opsets, find those with opset closest to the requested opset but not higher

Reply:
Yep, I figured as such. Updated the logic to reflect this.